.github/actions/surefire-reports/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Archive Surefire reports
+description: It will upload and archive surefire reports per particular test run.
+inputs:
+  job-name:
+    description: 'The name of the particular job run.'
+    required: true
+  release-branches:
+    description: 'List of all related release branches (in JSON format)'
+    required: false
+    default: '["refs/heads/release/22.0"]'
+  path-pattern:
+    description: 'The path pattern from where to get the surefire reports.'
+    required: false
+    default: |
+      **/target/surefire-reports/TEST*.xml
+      **/**/target/surefire-reports/TEST*.xml
+      **/**/**/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/base/target/surefire-reports/TEST*.xml
+      quarkus/tests/integration/target/surefire-reports/TEST*.xml
+      testsuite/model/target/surefire-reports*/TEST*.xml
+      crypto/default/target/surefire-reports/TEST*.xml
+      crypto/fips1402/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/base-ui/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/webauthn/target/surefire-reports/TEST*.xml
+      testsuite/integration-arquillian/tests/other/sssd/target/surefire-reports/TEST*.xml
```

Comment:
Not super keen on having to manually list all these patterns. 

Is `**` only matching a single directory, or does it match a hierarchy? If the latter then just `**/target/surefire-reports/TEST*.xml` should include everything?

Alternatively if that doesn't work could we use `find -name surefire-reports` to build the list?

Could we build this list with `find -name surefire-reports` for example, or

Reply:
I will test it and let you know

Reply:
@stianst comments addressed -> there are 3 builds which fails -> don't know why, get back on it on Monday (will be off)